pkg/mimirpb/timeseries_test.go

```diff
@@ -194,3 +197,216 @@ func TestDeepCopyTimeseriesExemplars(t *testing.T) {
 	// dst1 should use much smaller buffer than dst2.
 	assert.Less(t, cap(*dst1.yoloSlice), cap(*dst2.yoloSlice))
 }
+
+func TestPreallocTimeseries_Unmarshal(t *testing.T) {
+	// Prepare message
+	msg := PreallocTimeseries{}
+	{
+		src := PreallocTimeseries{
+			TimeSeries: &TimeSeries{
+				Labels: []LabelAdapter{
+					{Name: "sampleLabel1", Value: "sampleValue1"},
+					{Name: "sampleLabel2", Value: "sampleValue2"},
+				},
+				Samples: []Sample{
+					{Value: 1, TimestampMs: 2},
+					{Value: 3, TimestampMs: 4},
+				},
+			},
+		}
+
+		data, err := src.Marshal()
+		require.NoError(t, err)
+
+		require.NoError(t, msg.Unmarshal(data))
+		require.True(t, src.Equal(msg.TimeSeries))
+		require.NotNil(t, msg.unmarshalData)
+	}
+
+	correctMarshaledData := make([]byte, len(msg.unmarshalData))
+	copy(correctMarshaledData, msg.unmarshalData)
+
+	// Set cached version to random bytes (make a new slice, because labels in TimeSeries use the original byte slice).
+	msg.unmarshalData = make([]byte, 100)
```

Comment:
[nit] To have an higher confidence on the assertions below, I would store this slice in another variable too (e.g. `cachedData`) and then use it as expectation. This guarantees that we're not asserting on yet another version of `msg.unmarshalData` which could be overwritten when calling one of the functions on `msg` below.

Reply:
Added.